ui/src/App.tsx

```diff
@@ -1,19 +1,45 @@
 import React from 'react';
-import logo from './logo.svg';
 import './App.css';
 import TextUpload from './TextUpload';
 import ImageUpload from './ImageUpload';
 
-const App = () => {
-  return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <TextUpload />
-        <ImageUpload/>
-      </header>
-    </div>
-  );
+class App extends React.Component<any, any> {
+  constructor(props: any) {
+    super(props);
+    this.state = {topText: '', bottomText: ''}
+  }
+
+  getTopText = (text: String) => {
+    let newState = this.state;
+    newState = {
+      topText: text,
+      bottomText: this.state.bottomText
+    }
```

Comment:
Much of this is unnecessary.  If only changing the `topText` value, that's all you need to send to `setState`.  No need to send both values.